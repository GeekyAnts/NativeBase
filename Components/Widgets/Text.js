/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

	prepareRootProps() {

		var type = { 
			color: this.getTheme().textColor,
			fontSize: this.getTheme().fontSizeBase,
			lineHeight: this.getTheme().lineHeight
		}

		var defaultProps = {
			style: type
		}	  

		return computeProps(this.props, defaultProps);

	}
	render() {
		return(
			<Text {...this.prepareRootProps()}>{this.props.children}</Text>
		);
	}    

}

