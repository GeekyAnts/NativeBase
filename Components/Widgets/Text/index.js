/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../../Base/NativeBaseComponent';
import computeProps from '../../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

  propTypes: {
        style : React.PropTypes.object
    }

  prepareRootProps() {

    var type = {
      color: this.getContextForegroundColor(),
      fontSize: this.getTheme().fontSizeBase,
      lineHeight: this.getTheme().lineHeight,
      fontFamily: this.getTheme().fontFamily,
      backgroundColor: this.getTheme().textBgColor
    }

    var defaultProps = {
      style: type
    }

    return computeProps(this.props, defaultProps);

  }
  render() {
    return(
      <Text ref={c => this._root = c} {...this.prepareRootProps()}>{this.props.children}</Text>
    );
  }

}
