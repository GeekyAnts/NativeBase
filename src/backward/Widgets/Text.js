/* @flow */


import React from 'react';
import { Text as TextRN } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import NativeBaseComponent from '../Base/NativeBaseComponent';


class Text extends NativeBaseComponent {

  render() {
    return (
      <TextRN ref={c => this._root = c} {...this.props}>{this.props.children}</TextRN>
    );
  }

}

Text.propTypes = {
  ...TextRN.propTypes,
  style: React.PropTypes.object,
};

const StyledText = connectStyle('NativeBase.Text', {}, mapPropsToStyleNames)(Text);

export {
  StyledText as Text,
};
