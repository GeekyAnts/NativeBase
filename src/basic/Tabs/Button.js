const React = require("react");
const ReactNative = require("react-native");
const { TouchableOpacity, View } = ReactNative;

const Button = props => {
  return (
    <TouchableOpacity {...props}>
      {props.children}
    </TouchableOpacity>
  );
};

module.exports = Button;
//export default Button;
