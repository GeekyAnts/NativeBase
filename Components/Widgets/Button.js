/* @flow */
'use strict';

import React from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';
import IconNB from './Icon';
import Icon from './Icon';
import Text from './Text';
import _ from 'lodash';


export default class Button extends NativeBaseComponent {

    propTypes: {
        style : React.PropTypes.object,
        textStyle : React.PropTypes.object,
        block : React.PropTypes.bool,
        primary : React.PropTypes.bool,
        transparent : React.PropTypes.bool,
        success : React.PropTypes.bool,
        danger : React.PropTypes.bool,
        warning : React.PropTypes.bool,
        info : React.PropTypes.bool,
        bordered : React.PropTypes.bool,
        disabled : React.PropTypes.bool,
        rounded : React.PropTypes.bool,
        large : React.PropTypes.bool,
        small : React.PropTypes.bool,
        inputButton : React.PropTypes.bool
    }

    getInitialStyle() {
        return {
            button: {
                paddingVertical: (this.props.vertical) ? 0 :this.getTheme().buttonPadding,
                paddingHorizontal: (this.props.vertical) ? 5 :this.getTheme().buttonPadding+2,
                justifyContent: (this.props.block) ? 'center' : 'space-around',
                flexDirection:  (this.props.vertical) ? 'column' : 'row',
                alignSelf: 'center',
                alignItems: 'center',
                backgroundColor: this.getTheme().btnPrimaryBg,
                elevation: (this.props.transparent || this.props.bordered) ? 0 : 3,
                shadowColor: (this.props.transparent || this.props.bordered) ? undefined : '#000',
                shadowOffset: (this.props.transparent || this.props.bordered) ? undefined : {width: 0, height: 2},
                shadowOpacity: (this.props.transparent || this.props.bordered) ? undefined : 0.2,
                shadowRadius: (this.props.transparent || this.props.bordered) ? undefined : 2
            }
        }
    }

    prepareRootProps() {

        var type = {
            backgroundColor:    (this.props.primary) ? this.getTheme().btnPrimaryBg :
                                ((this.props.transparent) || (this.props.bordered)) ? 'rgba(0,0,0,0)' :
                                (this.props.success) ? this.getTheme().btnSuccessBg :
                                (this.props.danger) ? this.getTheme().btnDangerBg :
                                (this.props.warning) ? this.getTheme().btnWarningBg :
                                (this.props.info) ? this.getTheme().btnInfoBg :
                                (this.props.backgroundColor) ? this.props.backgroundColor :
                                (this.props.disabled) ? this.getTheme().btnDisabledBg :
                                this.getInitialStyle().button.backgroundColor,
            borderRadius: (this.props.rounded) ? this.getTheme().borderRadiusLarge : this.getTheme().borderRadiusBase,
            borderWidth: (this.props.bordered) ? 1 : 0,
            borderColor:    (this.props.primary) ? this.getTheme().primary :
                            (this.props.success) ? this.getTheme().btnSuccessBg :
                            (this.props.danger) ? this.getTheme().btnDangerBg :
                            (this.props.warning) ? this.getTheme().btnWarningBg :
                            (this.props.info) ? this.getTheme().btnInfoBg :
                            this.getInitialStyle().button.backgroundColor,
            height: (this.props.large) ? 60 : (this.props.small) ? 35 : 38,
            alignSelf: (this.props.block) ? 'stretch' : 'flex-start'
        }

        var addedProps = _.merge(this.getInitialStyle().button,type);

        var defaultProps = {
            style: addedProps,
            capitalize: true
        }

        return computeProps(this.props, defaultProps);
    }

    getTextStyle() {

        var mergedStyle = {};
        var btnType = {
            fontFamily: this.getTheme().btnFontFamily,
            marginLeft: (this.props.vertical) ? 0 : (this.iconPresent() && !this.props.iconRight) ? this.getTheme().iconMargin : 0,
            marginRight: (this.iconPresent() && this.props.iconRight) ? this.getTheme().iconMargin : 0,
            color:
                    ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                    ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                    ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                    ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                    ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                    ((this.props.bordered)) ? this.getTheme().btnPrimaryBg :
                    (this.props.color) ? this.props.color :
                    (this.props.transparent) ? this.getContextForegroundColor() :
                    this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().btnTextSizeLarge : (this.props.small) ? this.getTheme().btnTextSizeSmall : this.getTheme().btnTextSize,

            lineHeight: (this.props.large) ? 29 : (this.props.small) ? 16 : this.getTheme().btnLineHeight
        }

        return _.merge(mergedStyle, btnType, this.props.textStyle);
    }

    getIconProps(icon) {

        var defaultStyle = {
            color:
                ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                (this.props.bordered) ? this.getTheme().btnPrimaryBg :
                (this.props.color) ? this.props.color :
                (this.props.header) ? this.getTheme().toolbarTextColor :
                (this.props.transparent) ? this.getContextForegroundColor() :
                this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().iconSizeLarge : (this.props.small) ? this.getTheme().iconSizeSmall : (this.props.inputButton) ? this.getTheme().toolbarIconSize : this.getTheme().iconFontSize-5,
            lineHeight: (this.props.large) ? 52: (this.props.small || this.props.inputButton) ? 22 : this.getTheme().iconLineHeight-9
        }

        var defaultProps = {
            style: defaultStyle
        }

        return computeProps(icon.props, defaultProps);
    }
    iconPresent() {
        var iconComponentPresent = false;
        React.Children.forEach(this.props.children, function (child) {
            if(child.type == Icon)
                iconComponentPresent = true;
        })
        return iconComponentPresent;
    }
    renderChildren() {
        
        if(typeof this.props.children == 'string') {
            return <Text style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>
        }

        else if(this.props.children.type == IconNB) {
            return React.cloneElement(this.props.children, this.getIconProps(this.props.children));
        }

        else if(Array.isArray(this.props.children)) {
            var newChildren = [];

            var childrenArray = React.Children.toArray(this.props.children);

            var iconElement = [];
            iconElement = _.remove(childrenArray, function(item) {
                if(item.type == IconNB) {
                    return true;
                }
            });

            if(this.props.iconRight) {
                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }

                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);
            }

            else if(this.props.iconLeft || iconElement.length>0) {
                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);

                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }
            }
            else {
                return <Text style={this.getTextStyle()}>{(Platform.OS==='ios') ? this.props.children : this.props.children.toUpperCase()}</Text>
            }

            return newChildren;

        }

        else
            return React.cloneElement(this.props.children);

    }

    render() {
        return(
            <TouchableOpacity {...this.prepareRootProps()} >
                {this.renderChildren()}
            </TouchableOpacity>
        );
    }
}
