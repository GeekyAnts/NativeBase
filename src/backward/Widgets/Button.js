/* @flow */


import React, { Component } from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import _ from 'lodash';
import variables from '../../theme/variables';
import { Icon } from './Icon';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import computeProps from '../../Utils/computeProps';
import { Text } from './Text';


class Button extends Component {


  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: (this.props.bordered) ? 1 : undefined,
        borderRadius: (this.props.rounded && this.props.bordered) ? variables.borderRadiusLarge : 2,
      },
    };
  }

  iconPresent() {
    let iconComponentPresent = false;
    React.Children.forEach(this.props.children, (child) => {
      if (child.type === Icon) { iconComponentPresent = true; }
    });
    return iconComponentPresent;
  }

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn,
    };

    return computeProps(this.props, defaultProps);
  }

  renderChildren() {
    if (typeof this.props.children === 'string') {
      return <Text style={this.props.textStyle}>{(Platform.OS === 'ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>;
    } else if (this.props.children.type === Icon) {
      return React.cloneElement(this.props.children);
    } else if (Array.isArray(this.props.children)) {
      const newChildren = [];

      const childrenArray = React.Children.toArray(this.props.children);

      let iconElement = [];
      iconElement = _.remove(childrenArray, (item) => {
        if (item.type === Icon) {
          return true;
        }
        return null;
      });

      if (this.props.iconRight) {
        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }

        newChildren.push(React.cloneElement(iconElement[0]));
      } else if (this.props.iconLeft || iconElement.length > 0) {
        newChildren.push(React.cloneElement(iconElement[0]));

        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }
      } else {
        return <Text>{this.props.children}</Text>;
      }

      return newChildren;
    }
    return React.cloneElement(this.props.children);
  }

  render() {
    return (
      <TouchableOpacity
        ref={(c) => { this._root = c; }}
        {...this.prepareRootProps()} activeOpacity={0.5}
      >
        {this.renderChildren()}
      </TouchableOpacity>
    );
  }
}


Button.propTypes = {
  ...TouchableOpacity.propTypes,
  style: React.PropTypes.object,
  textStyle: React.PropTypes.object,
  block: React.PropTypes.bool,
  primary: React.PropTypes.bool,
  transparent: React.PropTypes.bool,
  success: React.PropTypes.bool,
  danger: React.PropTypes.bool,
  warning: React.PropTypes.bool,
  info: React.PropTypes.bool,
  bordered: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  rounded: React.PropTypes.bool,
  large: React.PropTypes.bool,
  small: React.PropTypes.bool,
  inputButton: React.PropTypes.bool,
  tabButton: React.PropTypes.bool,
};


const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export {
  StyledButton as Button,
};
