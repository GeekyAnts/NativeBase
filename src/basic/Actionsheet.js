/* eslint-disable no-use-before-define */
/* eslint-disable radix */

import React, { Component } from 'react';
import {
  ActionSheetIOS,
  Dimensions,
  FlatList,
  Modal,
  Platform,
  TouchableOpacity,
  StyleSheet,
  ViewPropTypes,
} from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';
import commonColor, { PLATFORM } from '../theme/variables/commonColor';

import { Text } from './Text';
import { Icon } from './Icon';
import { Left } from './Left';
import { Right } from './Right';
import { Body } from './Body';
import { ListItem } from './ListItem';

class ActionSheetContainer extends Component {
  static show(config, callback) {
    this.actionsheetInstance._root.showActionSheet(config, callback);
  }
  static hide() {
    this.actionsheetInstance._root.hideActionSheet();
  }

  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      items: [],
    };
  }

  componentDidMount() {
    if (!this.props.autoHide && this.props.duration) {
      // eslint-disable-next-line no-console
      console.warn(`It's not recommended to set autoHide false with duration`);
    }
  }

  static actionsheetInstance;

  showActionSheet(config, callback) {
    if (Platform.OS === PLATFORM.IOS) {
      if (typeof config.options[0] === 'object') {
        const options = config.options;
        const filtered = options.map((item) => {
          return item.text;
        });

        const filteredConfig = { ...config, options: filtered };
        ActionSheetIOS.showActionSheetWithOptions(filteredConfig, callback);
      } else {
        ActionSheetIOS.showActionSheetWithOptions(config, callback);
      }
    } else {
      this.setState({
        items: config.options,
        title: config.title,
        message: config.message,
        destructiveButtonIndex: config.destructiveButtonIndex,
        cancelButtonIndex: config.cancelButtonIndex,
        modalVisible: true,
        callback,
        style: config.style,
        fontStyle: config.fontStyle,
      });
    }
  }

  hideActionSheet() {
    this.setState({ modalVisible: false });
  }

  render() {
    return (
      <Modal
        animationType={'fade'}
        transparent
        visible={this.state.modalVisible}
        onRequestClose={() => {
          this.state.callback(this.state.cancelButtonIndex);
          this.setState({ modalVisible: false });
        }}
      >
        <TouchableOpacity
          activeOpacity={1}
          onPress={() => {
            this.state.callback(this.state.cancelButtonIndex);
            this.setState({ modalVisible: false });
          }}
          style={styles.containerTouchable}
        >
          <TouchableOpacity
            activeOpacity={1}
            style={[styles.innerTouchable, this.state.style]}
          >
            {this.state.title ? (
              <Text style={styles.touchableText}>{this.state.title}</Text>
            ) : null}
            <FlatList
              style={[
                styles.flatList,
                { marginTop: this.state.title ? commonColor.marginTop : 0 },
              ]}
              data={this.state.items}
              keyExtractor={(item, index) => String(index)}
              renderItem={({ index, item }) => {
                return typeof this.state.items[0] === 'string' ? (
                  <ListItem
                    onPress={() => {
                      this.state.callback(parseInt(index));
                      this.setState({ modalVisible: false });
                    }}
                    style={styles.listItem}
                  >
                    <Text style={this.state.fontStyle}>{item}</Text>
                  </ListItem>
                ) : (
                  <ListItem
                    onPress={() => {
                      this.state.callback(parseInt(index));
                      this.setState({ modalVisible: false });
                    }}
                    style={[
                      styles.listItem,
                      {
                        height: commonColor.listItemHeight,
                      },
                    ]}
                    icon
                  >
                    <Left>
                      <Icon
                        name={item.icon}
                        type={item.iconType}
                        style={{
                          color: item.iconColor ? item.iconColor : undefined,
                        }}
                      />
                    </Left>
                    <Body style={styles.listItemBody}>
                      <Text style={this.state.fontStyle}>{item.text}</Text>
                    </Body>
                    <Right />
                  </ListItem>
                );
              }}
            />
          </TouchableOpacity>
        </TouchableOpacity>
      </Modal>
    );
  }
}

ActionSheetContainer.propTypes = {
  ...ViewPropTypes,
};

const styles = StyleSheet.create({
  containerTouchable: {
    backgroundColor: commonColor.containerTouchableBackgroundColor,
    flex: 1,
    justifyContent: 'flex-end',
  },
  flatList: {
    marginHorizontal: commonColor.marginHorizontal,
  },
  innerTouchable: {
    backgroundColor: commonColor.innerTouchableBackgroundColor,
    minHeight: commonColor.minHeight,
    maxHeight: Dimensions.get('window').height / 2,
    padding: commonColor.padding,
    elevation: commonColor.elevation,
  },
  listItem: {
    borderColor: commonColor.listItemBorderColor,
    marginLeft: commonColor.marginLeft,
  },
  listItemBody: {
    borderColor: commonColor.listItemBorderColor,
    paddingLeft: commonColor.marginLeft / 2,
  },
  touchableText: {
    color: commonColor.touchableTextColor,
  },
});

const StyledActionSheetContainer = connectStyle(
  'NativeBase.ActionSheetContainer',
  {},
  mapPropsToStyleNames
)(ActionSheetContainer);

export { StyledActionSheetContainer as ActionSheetContainer };
