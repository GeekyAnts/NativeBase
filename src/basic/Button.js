/* eslint-disable new-cap */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  TouchableOpacity,
  Platform,
  View,
  TouchableNativeFeedback,
  StyleSheet
} from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import variable from '../theme/variables/platform';
import { PLATFORM } from '../theme/variables/commonColor';
import computeProps from '../utils/computeProps';
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';

import { Text } from './Text';

class Button extends Component {
  static contextTypes = {
    theme: PropTypes.object
  };

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: this.props.bordered
          ? variable.buttonDefaultBorderWidth
          : undefined,
        borderRadius:
          this.props.rounded && this.props.bordered
            ? variable.borderRadiusLarge
            : variable.buttonDefaultBorderRadius
      }
    };
  }

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn
    };

    if (Array.isArray(this.props.style)) {
      const flattenedStyle = this.props.style.reduce(
        (accumulator, currentValue) => accumulator.concat(currentValue),
        []
      );
      return computeProps(
        { ...this.props, style: flattenedStyle },
        defaultProps
      );
    }

    return computeProps(this.props, defaultProps);
  }
  render() {
    const variables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable;
    const children =
      Platform.OS === PLATFORM.IOS
        ? this.props.children
        : React.Children.map(this.props.children, child =>
            child && child.type === Text
              ? React.cloneElement(child, {
                uppercase: this.props.buttonUppercaseAndroidText === false
                ? false : variables.buttonUppercaseAndroidText,
                ...child.props
              })
              : child
          );
    if (
      Platform.OS === PLATFORM.IOS ||
      Platform.OS === PLATFORM.WEB ||
      variables.androidRipple === false ||
      Platform.Version < 21
    ) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={c => (this._root = c)}
          activeOpacity={
            this.props.activeOpacity > 0
              ? this.props.activeOpacity
              : variable.buttonDefaultActiveOpacity
          }
        >
          {children}
        </TouchableOpacity>
      );
    }
    if (this.props.rounded) {
      const buttonStyle = { ...this.prepareRootProps().style };
      const buttonFlex =
        this.props.full || this.props.block
          ? variable.buttonDefaultFlex
          : buttonStyle.flex;
      return (
        <View
          style={[
            { maxHeight: buttonStyle.height },
            buttonStyle,
            { paddingTop: undefined, paddingBottom: undefined }
          ]}
        >
          <TouchableNativeFeedback
            ref={c => (this._root = c)}
            background={TouchableNativeFeedback.Ripple(
              this.props.androidRippleColor || variables.androidRippleColor,
              true
            )}
            {...this.prepareRootProps()}
          >
            <View
              style={[
                // eslint-disable-next-line no-use-before-define
                styles.childContainer,
                {
                  paddingTop: buttonStyle.paddingTop,
                  paddingBottom: buttonStyle.paddingBottom,
                  height: buttonStyle.height,
                  flexGrow: buttonFlex
                }
              ]}
            >
              {children}
            </View>
          </TouchableNativeFeedback>
        </View>
      );
    }
    return (
      <TouchableNativeFeedback
        ref={c => (this._root = c)}
        onPress={this.props.onPress}
        background={
          this.props.transparent
            ? TouchableNativeFeedback.Ripple('transparent')
            : TouchableNativeFeedback.Ripple(
                variables.androidRippleColor,
                false
              )
        }
        {...this.prepareRootProps()}
      >
        <View {...this.prepareRootProps()}>{children}</View>
      </TouchableNativeFeedback>
    );
  }
}

Button.propTypes = {
  ...TouchableOpacity.propTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ]),
  block: PropTypes.bool,
  primary: PropTypes.bool,
  transparent: PropTypes.bool,
  success: PropTypes.bool,
  danger: PropTypes.bool,
  warning: PropTypes.bool,
  info: PropTypes.bool,
  bordered: PropTypes.bool,
  disabled: PropTypes.bool,
  rounded: PropTypes.bool,
  large: PropTypes.bool,
  small: PropTypes.bool,
  active: PropTypes.bool
};

const styles = StyleSheet.create({
  childContainer: {
    flexShrink: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

const StyledButton = connectStyle(
  'NativeBase.Button',
  {},
  mapPropsToStyleNames
)(Button);
export { StyledButton as Button };
