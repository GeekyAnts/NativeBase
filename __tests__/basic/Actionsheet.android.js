import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Button } from "./../../src/basic/Button";
import { ActionSheet } from "./../../src/basic/Actionsheet";
import { Text } from "./../../src/basic/Text";

var REGULAR_BUTTONS = ["Option 0", "Option 1", "Option 2", "Delete", "Cancel"];
var DESTRUCTIVE_INDEX = 3;
var CANCEL_INDEX = 4;
var ICON_BUTTONS = [
  { text: "Option 0", icon: "american-football", iconColor: "#2c8ef4" },
  { text: "Option 1", icon: "analytics", iconColor: "#f42ced" },
  { text: "Option 2", icon: "aperture", iconColor: "#ea943b" },
  { text: "Delete", icon: "trash", iconColor: "#fa213b" },
  { text: "Cancel", icon: "close", iconColor: "#25de5b" }
];

// Note: test renderer must be required after react-native.

jest.mock("Platform", () => {
  const Platform = require.requireActual("Platform");
  Platform.OS = "android";
  return Platform;
});

it("renders Regular ActionSheet", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          ActionSheet.show(
            {
              options: REGULAR_BUTTONS,
              cancelButtonIndex: CANCEL_INDEX,
              destructiveButtonIndex: DESTRUCTIVE_INDEX,
              title: "Select an option"
            },
            buttonIndex => {
              this.setState({ clicked: REGULAR_BUTTONS[buttonIndex] });
            }
          )
        }
      >
        <Text>Actionsheet</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Icon ActionSheet", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          ActionSheet.show(
            {
              options: ICON_BUTTONS,
              cancelButtonIndex: CANCEL_INDEX,
              destructiveButtonIndex: DESTRUCTIVE_INDEX,
              title: "Select an option"
            },
            buttonIndex => {
              this.setState({ clicked: ICON_BUTTONS[buttonIndex] });
            }
          )
        }
      >
        <Text>Actionsheet</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
