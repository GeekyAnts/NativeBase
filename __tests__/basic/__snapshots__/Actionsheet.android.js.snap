// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Icon ActionSheet 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`renders Regular ActionSheet 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;
